### ISSUE TITLE
Tests fail if AWS_REGION is set

### ISSUE THREAD

**Issue by rifelpet** [kubernetes/kops#4339]
I noticed that a few tests fail if the AWS_REGION environment variable is set.
```
$ AWS_REGION=us-west-2 make test
--- FAIL: TestMinimalCloudformation (3.55s)
	integration_test.go:558: diff for key Resources.AWSAutoScalingLaunchConfigurationmasterustest1amastersminimalexamplecom.Properties.UserData:
		...
		+ export AWS_REGION=us-west-2
		...
	integration_test.go:559: cloudformation output differed from expected
=== RUN   TestAdditionalUserData
--- FAIL: TestAdditionalUserData (3.35s)
	integration_test.go:558: diff for key Resources.AWSAutoScalingLaunchConfigurationmasterustest1amastersadditionaluserdataexamplecom.Properties.UserData:
		...
		+ export AWS_REGION=us-west-2
		...
	integration_test.go:559: cloudformation output differed from expected
--- FAIL: TestBootstrapUserData (0.02s)
	bootstrapscript_test.go:152: case 0 failed, actual output differed from expected (tests/data/bootstrapscript_0.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 1 failed, actual output differed from expected (tests/data/bootstrapscript_0.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 2 failed, actual output differed from expected (tests/data/bootstrapscript_1.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 3 failed, actual output differed from expected (tests/data/bootstrapscript_2.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 4 failed, actual output differed from expected (tests/data/bootstrapscript_3.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 5 failed, actual output differed from expected (tests/data/bootstrapscript_4.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 6 failed, actual output differed from expected (tests/data/bootstrapscript_3.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
	bootstrapscript_test.go:152: case 7 failed, actual output differed from expected (tests/data/bootstrapscript_5.txt).
	bootstrapscript_test.go:153: diff:
		...
		+ export AWS_REGION=us-west-2
		...
```

It'd be great if we could add isolation to these tests so that their behavior isn't affected by environment variables that happen to be set in the parent shell.

**Comment by chrislovecnm** [comment]
Related https://github.com/kubernetes/kops/issues/2657

**Comment by rifelpet** [comment]
Closed via #4595
